doc/source/coding-style/pep8.rst

```diff
@@ -1,229 +1,148 @@
 
-Indentation should be used to emphasize:
+Multiple Imports
+~~~~~~~~~~~~~~~~
+You should place imports in separate lines unless they are modules from the same
+package.
 
- - Body of a control statement, such as a loop or a select statement
- - Body of a conditional statement
- - New scope block
+.. tabs::
 
-.. code:: python
+    .. tab:: Avoid
 
-   class MyFirstClass:
-       """MyFirstClass docstring"""
+        .. code-block:: python
+        
+           import math, sys
 
-   class MySecondClass:
-       """MySecondClass docstring"""
+           from my_package import my_module
+           from my_package import my_other_module
+        
 
-   def top_level_function():
-       """Top level function docstring"""
-       return
-
-For improved readability, add blank lines or wrapping lines. Two
-blank lines should be added before and after all function and class
-definitions.
-
-Inside a class, use a single line before any method definition.
-
-.. code:: python
-
-   class MyClass:
-       """MyClass docstring"""
-
-   def first_method(self):
-       """First method docstring"""
-       return
-
-   def second_method(self):
-       """Second method docstring"""
-       return
-
-To make it clear when a 'paragraph' of code is complete and a new section
-is starting, use a blank line to separate logical sections.
-
-Instead of:
-
-.. code::
-
-   if x < y:
-
-       STATEMENTS_A
-
-   else:
-
-       if x > y:
-
-           STATEMENTS_B
-
-       else:
-
-           STATEMENTS_C
-
-   if x > 0 and x < 10:
-
-       print("x is a positive single digit.")
-
-Use:
-
-.. code::
-
-   if x < y:
-       STATEMENTS_A
-   else:
-       if x > y:
-           STATEMENTS_B
-       else:
-           STATEMENTS_C
-
-   if x > 0 and x < 10:
-       print("x is a positive single digit.")
-   elif x < 0:
-       print("x is less than zero.")
-
-
-Maximum Line Length
--------------------
-For source code lines, best practice is to keep the length at or below
-100 characters. For docstrings and comments, best practice is to keep
-the length at or below 72 characters.
-
-Lines longer than these recommended limits might not display properly
-on some terminals and tools or might be difficult to follow. For example,
-this line is difficult to follow:
+           def compute_logbase8(x):
+               return math.log(8, x)
+    
+    .. tab:: Use
+    
+        .. code-block:: python
+        
+           import math
+           import sys
 
-.. code:: python
+           from my_package import my_module, my_other_module
+        
 
-   employee_hours = [schedule.earliest_hour for employee in self.public_employees for schedule in employee.schedules]
+           def compute_logbase8(x):
+               return math.log(8, x)
 
-The line can be rewritten as:
 
-.. code:: python
+Imports Namespace
+~~~~~~~~~~~~~~~~~
+You should avoid using wild cards in imports because doing so can make it
+difficult to detect undefined names.  For more information, see `Python
```

Comment:
```suggestion
You should avoid using wildcards in imports because doing so can make it
difficult to detect undefined names. For more information, see `Python
```